locations/jclouds/src/main/java/org/apache/brooklyn/location/jclouds/DefaultConnectivityResolver.java

```diff
@@ -342,7 +343,12 @@ void publishNetworks(NodeMetadata node, Entity entity) {
     protected Iterable<HostAndPort> getReachableAddresses(NodeMetadata node, Predicate<? super HostAndPort> reachablePredicate, Duration timeout) {
         if (timeout == null) timeout = Duration.FIVE_MINUTES;
         Iterable<String> candidates = getResolvableAddressesWithMode(node);
-        return JcloudsUtil.getReachableAddresses(candidates, node.getLoginPort(), timeout, reachablePredicate);
+        Iterable<HostAndPort> result = JcloudsUtil.getReachableAddresses(candidates, node.getLoginPort(), timeout, reachablePredicate);
+        if (FluentIterable.from(result).filter(Predicates.equalTo(Optional.absent())).isEmpty()) {
+            LOG.error("None of the addresses {} of node {} are reachable in mode {}", new Object[]{Iterables.toString(candidates), node, getNetworkMode()});
+            throw new IllegalStateException("Could not find any reachable address for node: " + node + " in mode: " + getNetworkMode());
```

Comment:
Why not change the existing message on line 215?

Reply:
I think it is to avoid 185 to be executed, but we could change 215 as well

Reply:
If you want to avoid that I think it would be cleaner to wrap 185-203 in an `if (!Iterables.isEmpty(managementCandidates))`

Reply:
ok, although we need to check for `!FluentIterable.from(managementCandidates).filter(Predicates.equalTo(Optional.absent())).isEmpty()` as managementCandidates can't be empty